CMakeLists.txt

```diff
@@ -33,6 +33,7 @@ if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.19)
   cmake_policy(SET CMP0110 NEW)
 endif ()
 
+set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY ON)
```

Comment:
Maybe add a comment like
```
# Disable `make install` depending on `make all` since we want to control what we install more closely.
# E.g. installing tests is not really necessary so we might not have `install` depend on the tests.
```